types/react-native/index.d.ts

```diff
@@ -612,6 +613,15 @@ export interface FlexStyle {
  */
 export interface LayoutProps extends FlexStyle {}
 
+/**
+ * We should be able to specify a shadow offset with only one
+ * dimension or with both dimensions.
+ */
+export type ShadowOffset =
+  | { height: number }
+  | { width: number }
+  | { height: number, width: number };
```

Comment:
There's no need for Union type here, both `width` and `height` are optional values on the interface.
https://github.com/facebook/react-native/blob/c6b96c0df789717d53ec520ad28ba0ae00db6ec2/Libraries/StyleSheet/StyleSheetTypes.js#L120